File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports System.Globalization

Namespace GridViewLocalizeColumnCaptionMvc.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		<HttpPost>
		Public Function Index(ByVal language As String) As ActionResult
			Session("CurrentCulture") = CultureInfo.GetCultureInfo(language)
			CustomLocalizationHelper.ApplyCurrentCulture()
			Return View()
		End Function

		Public Function GridViewPartial() As ActionResult
			CustomLocalizationHelper.ApplyCurrentCulture()
			Return PartialView(Invoice.GetData())
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;
using System.Globalization;

namespace GridViewLocalizeColumnCaptionMvc.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        [HttpPost]
        public ActionResult Index(string language) {
            Session["CurrentCulture"] = CultureInfo.GetCultureInfo(language);
            CustomLocalizationHelper.ApplyCurrentCulture();
            return View();
        }

        public ActionResult GridViewPartial() {
            CustomLocalizationHelper.ApplyCurrentCulture();
            return PartialView(Invoice.GetData());
        }
    }
}